### PR TITLE
NUTCH-2284 Basic Authentication support for Nutch 2.X REST API.

### PR DESCRIPTION
**Pull request by kamaci**

### CODE REVIEW

**Comment by lewismc** [comment]
@kamaci can you please add JDK1.8 compliant Javadoc to the relevant method in NutchServer.java
 which states which properties from nutch-site.xml you need to set in order to use basic authentication?

**Comment by kamaci** [comment]
@lewismc I've committed the changes you wanted at your comment.

File: conf/nutch-default.xml [hunk 1]
```diff
@@ -1435,4 +1435,30 @@
   </description>
 </property>
 
+<property>
+  <name>restapi.auth</name>
+  <value>false</value>
+  <description>
+    Whether to enable HTTP basic authentication for communicating with RESTAPI.
+    Use the restapi.auth.username and restapi.auth.auth.password properties to configure
```
Comment (lewismc) [inline conf/nutch-default.xml]: Typo in restapi.auth.auth.password it should be restapi.auth.password

File: src/java/org/apache/nutch/api/NutchServer.java [hunk 2]
```diff
@@ -85,7 +88,12 @@
    * well as the logging granularity. If the latter option is not provided via
    * {@link org.apache.nutch.api.NutchServer#main(String[])} then it defaults to
    * 'INFO' however best attempts should always be made to specify a logging
-   * level.
+   * level.<br>
+   * {@link org.apache.nutch.api.NutchServer} can be run as secure. restapi.auth property
+   * should be set to true at <code>nutch-site.xml</code> to enable HTTP basic authentication
```
Comment (lewismc) [inline src/java/org/apache/nutch/api/NutchServer.java]: In JDK 1.8 I think the relevnt HTML encoding should be used e.g. &lt;code&gt;nutch-site.xml&lt;/code&gt;

File: conf/nutch-default.xml (hunk 1 above)
Comment (kamaci) [inline conf/nutch-default.xml]: @lewismc I've fixed typo.

**Comment by lewismc** [comment]
http://stackoverflow.com/questions/2898897/how-can-i-use-and-in-javadoc-without-formating

**Comment by kamaci** [comment]
I've updated escaped HTML characters at javadoc.

**Comment by kamaci** [comment]
@lewismc this PR is ready to check for merging.

**Pull request merged**